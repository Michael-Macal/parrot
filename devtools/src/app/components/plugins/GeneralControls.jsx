import React, { Component } from 'react';
import PropTypes from 'prop-types';

import TextField from 'material-ui/TextField';
import FlatButton from 'material-ui/FlatButton';
import RefreshIcon from 'material-ui/svg-icons/navigation/refresh';

const textareaStyle = {
  paddingLeft: 0,
};

class GeneralControls extends Component {
  static proptypes = {
    url: PropTypes.string.isRequired,
    setUrl: PropTypes.func.isRequired,
    onRefresh: PropTypes.func.isRequired,
  }

  setUrl = (event, url) => this.props.setUrl(url);

  render() {
    const { url, onRefresh } = this.props;
    return (
      <div>
        <TextField
          name="Parrot Server Host"
          floatingLabelText="Parrot Server Host"
          value={url}
          onChange={this.setUrl}
          textareaStyle={textareaStyle}
        />
        <FlatButton
          icon={<RefreshIcon />}
          onTouchTap={onRefresh}
        />
      </div>
    );
  }
}


export default GeneralControls;
