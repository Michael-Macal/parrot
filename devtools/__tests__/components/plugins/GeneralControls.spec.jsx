import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import GeneralControls from '../../../src/app/components/plugins/GeneralControls';

describe('GeneralControls component', () => {
  it('renders', () => {
    const props = {
      url: 'http://localhost:8080',
      setUrl: jest.fn(),
      onRefresh: jest.fn(),
    };
    const wrapper = shallow(React.createElement(GeneralControls, props));
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
