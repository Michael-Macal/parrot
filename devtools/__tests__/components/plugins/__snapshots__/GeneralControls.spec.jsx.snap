exports[`GeneralControls component renders 1`] = `
<div>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="Parrot Server Host"
    fullWidth={false}
    multiLine={false}
    name="Parrot Server Host"
    onChange={[Function]}
    rows={1}
    textareaStyle={
      Object {
        "paddingLeft": 0,
      }
    }
    type="text"
    underlineShow={true}
    value="http://localhost:8080" />
  <FlatButton
    disabled={false}
    icon={<NavigationRefresh />}
    labelPosition="after"
    labelStyle={Object {}}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    primary={false}
    secondary={false} />
</div>
`;
